lib/common.js

```diff
@@ -545,6 +574,7 @@ exports.readMysqlValue = function(parser, column, columnSchema) {
     case MysqlTypes.DECIMAL: // Deprecated in MySQL > 5.0.3
     case MysqlTypes.NEWDATE: // Not used
     default:
+      // TODO: Remove throw statement to make module safe
```

Comment:
I think that the performance penalty to wrap such a frequently called function in a ``try/catch`` could be significant. 

If there's an error parsing a column's value, how about we leave that column ``undefined?``  That allows you to differentiate between a ``null`` db value and a parsing error.

If there's an error parsing a row, halt parsing and emit a rowError with the raw format of the row and the error.

I believe taking a fail fast, exit early approach could work well.